contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -376,16 +376,16 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
             _payPrincipalWithoutTransfer(pool, amount.sub(accruedInterest));
         }
 
-        if (totalBorrowed(msg.sender) == 0) {
-            borrowingMutex.unlock(msg.sender);
-        }
-
         if (borrowed[pool][msg.sender] == 0) {
             nextInterestRepayTime[pool][msg.sender] = 0;
         }
+        pokeBorrowLimitTimer(pool, msg.sender);
 
         // transfer token from sender wallets
         _repay(pool, amount);
+        if (totalBorrowed(msg.sender) == 0) {
+            borrowingMutex.unlock(msg.sender);
+        }
```

Comment:
Moving these lines isn't strictly relevant to this PR, but it _is_ an important change to make.

The original version violates checks-effects-interactions and possibly allows a reentrancy attack. Unlocking the borrowing mutex is an outside interaction that should be executed only after all other transfers have completed, so that transferees don't have an opportunity to execute borrow function calls on an unlocked borrower.